src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.RuntimeIdentifierInference.targets

```
@@ -65,6 +65,8 @@ Copyright (c) .NET Foundation. All rights reserved.
                  (
                  '$(RuntimeIdentifier)' == '' and
                  '$(RuntimeIdentifiers)' == '' and
+                 '$(_IsExecutable)' == 'true' and
+                 '$(DisableImplicitRuntimeIdentifier)' != 'true' and
```

Per our discussion, I think this may be what we want.

```suggestion
                 '$(_IsExecutable)' == 'true' and '$(IsTestProject)' != 'true' and
                 '$(IsRidAgnostic)' != 'true' and
                 '$(DisableImplicitRuntimeIdentifier)' != 'true' and
```